src/types/response.ts

```diff
@@ -6,6 +6,33 @@ export enum ExecutionState {
   FAILED = "QUERY_STATE_FAILED",
 }
 
+export interface ExecutionResponseCSV {
+  data: string;
+  next_uri: string | null;
+  next_offset?: number;
+}
+
+export function concatResultCSV(
+  left: ExecutionResponseCSV,
+  right: ExecutionResponseCSV,
+): ExecutionResponseCSV {
+  left.next_uri = right.next_uri;
+  left.next_offset = right.next_offset;
+
+  // Trim last newline from the left and split both on newlines.
+  const leftData = left.data.trimEnd().split("\n");
+  const rightData = right.data.split("\n");
+  // Remove the header column from right:
+  rightData.shift();
+
+  return {
+    next_uri: right.next_uri,
+    next_offset: right.next_offset,
+    // Concatenate the two arrays and join them back into a single string
+    data: leftData.concat(rightData).join("\n"),
```

Comment:
there should be a way of doing:
```
const data = discardLines(right.data, 1) // consume header line, return the remaining
const total = stringConcat(left.data, right.data)
```

I just wonder how much this will amplify the memory consumption..
maybe this is a nitpick?